std/array.d

```diff
@@ -3927,3 +3927,213 @@ unittest
     assert(appS.data == "hellow");
     assert(appA.data == "hellow");
 }
+
+/++
+Params: a = The input elements
+
+Returns: A static array constructed from `a`.
+
+The type of elements can be specified implicitly (`int[2] a = [1,2].asStatic;`)
+or explicitly (`float[2] a = [1,2].asStatic!float`). When `a` is a range,
+the number of elements has to be given as template argument (eg `2.iota.asStatic!2`).
+Size and type can be combined (eg: `2.iota.asStatic!(byte[2])`).
+Range `a` can also be specified as a template argument (eg: `asStatic!(2.iota)`
+or `asStatic!(double, 2.iota)`).
+
+Note: `foo([1, 2, 3].asStatic)` may be inefficient because of the copies involved.
++/
+pragma(inline, true) T[n] asStatic(T, size_t n)(auto ref T[n] a) nothrow @safe pure @nogc
+{
+    return a;
+}
+
+/// ditto
+U[n] asStatic(U, T, size_t n)(auto ref T[n] a) nothrow @safe pure @nogc
+if (!is(U == T))
+{
+    import std.conv : emplaceRef;
+    U[n] ret = void;
+    static foreach (i; 0 .. n)
+    {
+        emplaceRef!U(ret[i], cast(U) a[i]);
+    }
+    return ret;
+}
+
+///
+nothrow pure @safe unittest
+{
+    auto a = [0, 1].asStatic;
+    assert(is(typeof(a) == int[2]) && a == [0, 1]);
```

Comment:
static

Reply:
done (for cases in public unittest only since assert/writeln pipeline only applies to these)